apps/idos-enclave/src/pages/methods/LitProtocol.tsx

```diff
@@ -0,0 +1,115 @@
+import { Lit } from "@idos-network/idos-sdk";
+import { encode as fromUintToString } from "@stablelib/base64";
+import { difference } from "lodash-es";
+import React, { useEffect, useMemo, useState } from "preact/compat";
+import { Paragraph } from "../../components/Paragraph";
+//@ts-ignore
+import { Enclave } from "../../lib/enclave";
+import type { MethodProps } from "./Chooser";
+
+type LoadingMsg = "initializingLit" | "generatingKey" | "storingKey" | "retrievingKey";
+
+export interface StorableData<T = string> {
+  key: string;
+  value: T;
+}
+
+const loadingMsgSrc: Record<LoadingMsg, string> = {
+  initializingLit: "Initializing Lit protocol...",
+  generatingKey: "Generating a key for you...",
+  retrievingKey: "Retriving stored key...",
+  storingKey: "Storing the generated key...",
+} as const;
+
+export default function LitProtocol({ onSuccess, store }: MethodProps) {
+  const [laodingProperty, setLoadingProperty] = useState<LoadingMsg>("initializingLit");
+  const litInstance = useMemo(() => new Lit("ethereum", store), [store]);
+
+  const ciphertext = store.get("lit-cipher-text");
+  const dataToEncryptHash = store.get("lit-data-to-encrypt-hash");
+
+  const startLoading = (newLoadingProp: LoadingMsg) => setLoadingProperty(newLoadingProp);
+
+  const updateStoredCipherAndHash = (cipher: string, dataHash: string) => {
+    store.set("lit-cipher-text", cipher);
+    store.set("lit-data-to-encrypt-hash", dataHash);
+  };
+
+  // biome-ignore lint/style/useDefaultParameterLast: <explanation>
+  const retriveLitKey = async (
+    prevUserWallets = [],
+    newUserWallets = [],
+    walletsChanged = false,
+  ) => {
+    startLoading("retrievingKey");
+    const key = await litInstance.decrypt(ciphertext, dataToEncryptHash, prevUserWallets);
+
+    if (walletsChanged) {
+      // re-encrypt here and removing previous signature info. new encryption means previous signature wont work
+      localStorage.removeItem("lit-session-key");
```

Comment:
@ditoglez this is a value that lit store on local storage on itself. removing this will trigger the user signature to be asked again. just to make sure it's the right signature since it's another wallets used. but I think ur right I will test it using same signature even if wallets changed